From 508121cfde12d0e35716bb7e524dc1f80c82262c Mon Sep 17 00:00:00 2001
From: Calvin Johnson <calvin.johnson@nxp.com>
Date: Tue, 20 Nov 2018 21:50:40 +0530
Subject: [PATCH] staging: fsl_ppfe/eth: misc clean up

- remove redundant hwfeature init
- remove unused vars from ls1012a_eth_platform_data
- To handle ls1012a errata_a010897, PPFE driver requires GUTS driver
to be compiled in. Select FSL_GUTS when PPFE driver is compiled.

Signed-off-by: Calvin Johnson <calvin.johnson@nxp.com>
---
 drivers/staging/fsl_ppfe/Kconfig   | 1 +
 drivers/staging/fsl_ppfe/pfe_eth.c | 3 ---
 drivers/staging/fsl_ppfe/pfe_eth.h | 4 ----
 3 files changed, 1 insertion(+), 7 deletions(-)

--- a/drivers/staging/fsl_ppfe/Kconfig
+++ b/drivers/staging/fsl_ppfe/Kconfig
@@ -3,6 +3,7 @@
 #
 config FSL_PPFE
 	bool "Freescale PPFE Driver"
+	select FSL_GUTS
 	default n
 	---help---
 	Freescale LS1012A SoC has a Programmable Packet Forwarding Engine.
--- a/drivers/staging/fsl_ppfe/pfe_eth.c
+++ b/drivers/staging/fsl_ppfe/pfe_eth.c
@@ -2437,9 +2437,6 @@ static int pfe_eth_init_one(struct pfe *
 	else
 		ndev->max_mtu = JUMBO_FRAME_SIZE_V2 - ETH_HLEN - ETH_FCS_LEN;
 
-	/* supported features */
-	ndev->hw_features = NETIF_F_SG;
-
 	/*Enable after checksum offload is validated */
 	ndev->hw_features = NETIF_F_RXCSUM | NETIF_F_IP_CSUM |
 		NETIF_F_IPV6_CSUM | NETIF_F_SG;
--- a/drivers/staging/fsl_ppfe/pfe_eth.h
+++ b/drivers/staging/fsl_ppfe/pfe_eth.h
@@ -37,10 +37,6 @@
 #define GEMAC_NO_PHY            BIT(0)
 
 struct ls1012a_eth_platform_data {
-	/* device specific information */
-	u32 device_flags;
-	char name[16];
-
 	/* board specific information */
 	u32 mii_config;
 	u32 phy_flags;
